### ISSUE TITLE
Custom equality tester doesn't work with #toThrow matcher

### ISSUE THREAD

**Issue by Vitaly-Rudenko** [jasmine/jasmine#1825]
Custom equality tester doesn't seem to work with #toThrow matcher, but works with #toEqual and #toBeRejectedWith matchers.

Is it intended behavior? Or am I using it incorrectly?

## Suite that reproduces the behavior (for bugs)
```javascript
class CustomError1 extends Error {}
class CustomError2 extends Error {}

describe('errorEqualityTester', () => {
    beforeAll(() => {
        jasmine.addCustomEqualityTester(
            (a, b) => {
                if (a instanceof Error && b instanceof Error) {
                    return (
                        a.constructor === b.constructor &&
                        a.message === b.message
                    );
                }
            }
        );
    });

    it('should work with toEqual()', () => {
        expect(new CustomError1('fake error'))
            .toEqual(new CustomError1('fake error'));

        expect(new CustomError1('fake error 1'))
            .not.toEqual(new CustomError1('fake error 2'));

        expect(new CustomError1('fake error'))
            .not.toEqual(new CustomError2('fake error'));
    });

    it('should work with toThrow()', () => {
        expect(() => { throw new CustomError1('fake error'); })
            .toThrow(new CustomError1('fake error'));

        expect(() => { throw new CustomError1('fake error 1'); })
            .not.toThrow(new CustomError1('fake error 2'));

        // fails
        expect(() => { throw new CustomError1('fake error'); })
            .not.toThrow(new CustomError2('fake error'));
    });

    it('should work with toBeRejectedWith()', async () => {
        await expectAsync(Promise.reject(new CustomError1('fake error')))
            .toBeRejectedWith(new CustomError1('fake error'));

        await expectAsync(Promise.reject(new CustomError1('fake error 1')))
            .not.toBeRejectedWith(new CustomError1('fake error 2'));

        await expectAsync(Promise.reject(new CustomError1('fake error')))
            .not.toBeRejectedWith(new CustomError2('fake error'));
    });
});
```
![image](https://user-images.githubusercontent.com/1491646/84365490-3514d780-abda-11ea-8af2-b57b81b0e17c.png)

## Your Environment
* Version used: 3.5.0
* Environment name and version: Node 10.15.1

**Comment by sgravrock** [comment]
It's a bug. It looks like it was fixed in dec67bd5 (your tests pass when I run them against that commit or newer) and the fix will ship with the next release.

**Comment by Vitaly-Rudenko** [comment]
> It's a bug. It looks like it was fixed in [dec67bd](https://github.com/jasmine/jasmine/commit/dec67bd535bfeeb9e4953f14c35ff56874aabcf9) (your tests pass when I run them against that commit or newer) and the fix will ship with the next release.

Yay! Thanks
